website/docs/crud.md

```diff
@@ -0,0 +1,294 @@
+
+`data` is type `dict`, key-values pair where key is the column name and value is the entry to be updated.
+
+### CRUDDataAccess
+This class is used to access data via `CRUD` and we have to extend this class to use `CRUD` in your application. This
+class is used to initialize `CRUD` and further used to access its functions.
+
+```python
+class CRUDDataAccess(DataAccess, CRUD):
+
+    def __init__(self, db_entity, db: SqlAlchemyDataHandlerRegistry, rule_validator: RuleValidator):
+        CRUD.__init__(self, db_entity, db, rule_validator)
+```
+`db_entity` is the database class or object to be passed for initialization.
+
+`db` is the `SqlAlchemyDataHandlerRegistry` instance used to connect to the database.
+
+`rule_validator` rules must be created with `RuleValidator` and passed.
+
+#### Functions provided by `CRUDDataAccess`
+
+- `get()` overrides the function in `CRUD` class, used to get data from database for a given `id`.
+```python
+def get(self, id: int):
+```
+`id` is the id of the column we want to query.
+
+
+- `delete()` overrides the function in `CRUD` class, deletes the data for the given `id`.
+```python
+def delete(self, id: int):
+```
+`id` is the id of the column to be deleted.
+
+#### Usage
+```python
+from core_lib.data_layers.data_access.db.crud.crud import CRUD
+from core_lib.data_layers.data_access.db.crud.crud_data_access import CRUDDataAccess
+from core_lib.rule_validator.rule_validator import ValueRuleValidator, RuleValidator
+from core_lib.data_layers.data.db.sqlalchemy.base import Base
+from core_lib.data_transform.result_to_dict import result_to_dict
+
+from sqlalchemy import Column, Integer, VARCHAR, Boolean
+
+class Data(Base):
+    __tablename__ = 'user_data'
+
+    id = Column(Integer, primary_key=True, nullable=False)
+    name = Column(VARCHAR(length=255), nullable=False, default="")
+    email = Column(VARCHAR(length=255), nullable=False, default="")
+    active = Column(Boolean, nullable=False, default="")
+
+
+class CRUDInit(CRUDDataAccess):
+    allowed_update_types = [
+        ValueRuleValidator('name', str),
+        ValueRuleValidator('email', str),
+        ValueRuleValidator('active', bool)
+    ]
+
+    rules_validator = RuleValidator(allowed_update_types)
+
+    def __init__(self):
+        CRUD.__init__(self, Data, db_handler, CRUDInit.rules_validator)
+
+crud = CRUDInit()
+
+crud.create({'name': 'Jon Doe', 'email': 'abc@def.com', 'active': True})
+
+data = result_to_dict(crud.get(1))
+print(data) # {'id': 1, 'name': 'Jon Doe', 'email': 'abc@def.com', 'active': True}
+
+crud.update(1, {'email': 'jon@doe.com'})
+
+data = result_to_dict(crud.get(1))
+print(data) # {'id': 1, 'name': 'Jon Doe', 'email': 'jon@doe.com', 'active': True}
+
+crud.delete(1)
+
+data = crud.get(1) # will raise StatusCodeException Not found
+```
+
+
+### CRUDSoftDeleteDataAccess
+This class is similar to `CRUDDataAccess` but is used to access and handle soft delete and soft deleted data.

```

Comment:
no need `this calss is`. just say 
similar to `CRUDDataAccess`. ...


`handle soft delete and soft deleted data.`. just say handle softdelete